### ISSUE TITLE
Fails to create results.csv

### ISSUE THREAD

**Issue by DrWilliamsSteven** [CancerRxGene/gdsctools#145]
Hi Thomas,

I've found that sometimes when I input a larger dataset (eg 400 drugs x 50 cell lines) it fails to create the results.csv file (and other files in INPUT and OUTPUT folders) and also fails to create some plots such as volcano_all_js.png.

Other individual associations and volcano plots seem to be fine.

Any suggestions appreciated. Let me know if you need me to supply any examples etc.

Thanks Steve Williams

**Comment by cokelaer** [comment]
Dear Steve, 

I believe that gdsctools shoudl scale up. The main example I used all the time is the what we call v17 data sete, which is included in gdsctools and has 265 drugs and about 1000 cell lines. More recently we used a new data set with 500 drugs and 1000 cell lines. So I think this is not the number of assocations that causes trouble. 

Which version of GDSCTools did you use ? It looks like this is a random fail or can you reproduce the errors ? 

Volcano plots are not always created: sometimes there is not valid tests. For instance if you have lots of NA for a given drug. It is probably not your case since you are speaking about the volcano_all plot but could you just check that the feasible tests are large:

    anova.diagnostics()

Otherwise, examples would be useful indeed to track down the issue although I would need a reproducible example. 

What platform do you use by the way ?
thanks

**Comment by cokelaer** [comment]
I've fixed a small bug I never encountered before which may be related to your problem: sometimes when computing the so-called signed-effects, it may be equal to zero but rounding errors lead to values slightly negative (e.g. -1e-16), which should have been rounded to zero. If not, a NAN was populated leading to issues in volcano plots. This may be related or not. In any case, this issue was fixed in version 0.16.1 now available on pypi.
